nat-lab/tests/utils/batching.py

```diff
@@ -0,0 +1,137 @@
+import asyncio
+import math
+import os
+import pytest
+import subprocess
+import typing
+from scapy.all import PcapReader, Packet  # type: ignore
+from typing import Callable, List, Optional
+
+
+def _generate_histogram(
+    data: list[int], buckets: int, bucket_size: int = 1
+) -> List[int]:
+    max_val = max(data)
```

Comment:
Maaybe some comment that you treat this `max` also for guarding against `len(data) == 0` would be nice, so someone  won't add `default=0` to it as a fix in the future

Reply:
Not sure if I understood correctly but `len(data) == 0` now happens at the call site. For all this function knows about - it expects non-zero data list. Do you mean maybe I should assert for `assert len(data) > 0`?

Reply:
Assert would be ok

Reply:
Surely, added it